forge-gui/res/cardsfolder/upcoming/champions_of_tyr.txt

```diff
@@ -0,0 +1,13 @@
+Name:Champions of Tyr
+ManaCost:2 W W
+Types:Creature Angel Knight
+PT:4/3
+K:Flying
+K:Double team
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigAddAPI | TriggerDescription$ When CARDNAME enters the battlefield, you get a boon with "When you cast your next creature spell, that creature enters the battlefield with your choice of a +1/+1 counter, a flying counter, or a lifelink counter on it."
+SVar:TrigAddAPI:DB$ Effect | RememberObjects$ TriggeredCard | Boon$ True | ForgetOnMoved$ Stack | ReplacementEffects$ ReplaceEnter
```

Comment:
Put the Trigger on the Effect because "When you cast your next etc" = trigger
Trigger can apply the replacement effect
Yeah it's a tangle, yay rules

Reply:
You can get the idea from Arcane Archery